### PR TITLE
fix: ChatAnthropicVertex returning None during structured output

### PR DESCRIPTION
**Pull request by kiarina**
When using `with_structured_output` with `ChatAnthropicVertex`, the server response is ignored, resulting in `None` being returned from the `invoke` method.

```python
import os
from langchain_google_vertexai.model_garden import ChatAnthropicVertex
from langchain_core.pydantic_v1 import BaseModel
from langchain_core.messages import HumanMessage

model = ChatAnthropicVertex(
    project=os.environ["PROJECT_ID"],
    location="us-east5",
    model="claude-3-opus@20240229"
)

class MyModel(BaseModel):
    name: str
    age: int

model_with_structured_output = model.with_structured_output(MyModel)
message = HumanMessage(content="My name is Erick and I am 27 years old")
print(model_with_structured_output.invoke([message]))
# None
```

The server response at this time is as follows:

```python
Message(id='msg_vrtx_01RuVwJ9ANqEjLKB2CciEz3r', content=[ToolUseBlock(id='toolu_vrtx_0147LC8SasuVZXi4RQsk1eow', input={'name': 'Erick', 'age': 27}, name='MyModel', type='tool_use')], model='claude-3-opus-20240229', role='assistant', stop_reason='tool_use', stop_sequence=None, type='message', usage=Usage(input_tokens=372, output_tokens=56))
```

In `_format_output`, the `content` of `AIMessage` becomes empty.

- https://github.com/langchain-ai/langchain-google/blob/main/libs/vertexai/langchain_google_vertexai/model_garden.py#L202-L226

```python
content=[] tool_calls=[{'name': 'MyModel', 'args': {'name': 'Erick', 'age': 27}, 'id': 'toolu_vrtx_0147LC8SasuVZXi4RQsk1eow'}]
```

Therefore, the condition in the following part of `ToolsOutputParser` does not retrieve data.

- https://github.com/langchain-ai/langchain-google/blob/main/libs/vertexai/langchain_google_vertexai/_anthropic_parsers.py#L30-L41

By making use of `AIMessage.tool_calls` when it is not empty, the issue is fixed.

After the fix, data can be retrieved as follows:

```python
name='Erick' age=27
```

### CODE REVIEW

**Review by lkuligin — APPROVED** [review]

**Comment by lkuligin** [comment]
Huge thanks for the fix!

**Comment by lkuligin** [comment]
@kiarina linter is still failing, could you take a look, please?

File: libs/vertexai/langchain_google_vertexai/_anthropic_parsers.py [hunk 1]
```diff
@@ -26,27 +26,20 @@ def parse_result(self, result: List[Generation], *, partial: bool = False) -> An
...
-        else:
-            content: List = message.content
-            _tool_calls = [dict(tc) for tc in _extract_tool_calls(content)]
-            # Map tool call id to index
-            id_to_index = {
-                block["id"]: i
-                for i, block in enumerate(content)
-                if block["type"] == "tool_use"
-            }
-            tool_calls = [{**tc, "index": id_to_index[tc["id"]]} for tc in _tool_calls]
```
Comment (kiarina) [inline libs/vertexai/langchain_google_vertexai/_anthropic_parsers.py:41]: I checked `mypy` for errors, but could not find where "index" was referenced, so I removed the whole thing.

**Comment by kiarina** [comment]
Sorry.
I had a different version of ruff, so I modified some unnecessary files.
After `poetry install --with lint`, I checked the modified files with `make lint` and undid the unnecessary modifications.
Locally, all checks passed.

**Comment by lkuligin** [comment]
@kiarina please, feel free to send a PR to the contribution guide if you find that some things were unclear to a new contributor!

**Pull request merged**